src/shared/components/NcFormBoxItem.vue

```diff
@@ -0,0 +1,190 @@
+<!--
+  - SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+-->
+
+<script setup lang="ts">
+import type { Slot } from 'vue'
+import type { VueClassType } from '../../utils/VueTypes.ts'
+
+import { useNcFormBox } from '../../components/NcFormBox/useNcFormBox.ts'
+import { createElementId } from '../../utils/createElementId.ts'
+import { isLegacy } from '../../utils/legacy.ts'
+
+defineOptions({ inheritAttrs: false })
+
+const {
+	tag = 'div',
+	label = undefined,
+	description = undefined,
+	invertedAccent = false,
+	class: rootClasses = undefined,
+	itemClasses = undefined,
+} = defineProps<{
+	/** Interactive item element's tag */
+	tag?: string
+	/** Main Label */
+	label?: string
+	/** Optional description below the label, also used for the aria-describedby */
+	description?: string
+	/** Accent on the description instead of the label */
+	invertedAccent?: boolean
+	/** Root element classes */
+	class?: VueClassType
+	/** Interactive item classes */
+	itemClasses?: VueClassType
+}>()
+
+defineEmits<{
+	/** Click on the item */
+	click: [event: MouseEvent]
+}>()
+
+const slots = defineSlots<{
+	/** Item's label custom content */
+	default?: Slot<{
+		/** IDRef of the description element if present */
+		descriptionId?: string
+	}>
+	/** Custom description content */
+	description?: Slot<{
+		/** IDRef of the description element if present */
+		descriptionId?: string
+	}>
+	/** Icon content */
+	icon?: Slot
+}>()
+
+const { formBoxItemClass } = useNcFormBox()
+
+const descriptionId = createElementId()
+const hasDescription = () => !!description || !!slots.description
+</script>
+
+<template>
+	<div
+		:class="[
+			rootClasses,
+			$style.formBoxItem,
+			formBoxItemClass,
+			{
+				[$style.formBoxItem_inverted]: invertedAccent && hasDescription(),
+				[$style.formBoxItem_legacy]: isLegacy,
+			},
+		]">
+		<span :class="$style.formBoxItem__content">
+			<component
+				:is="tag"
+				:class="[$style.formBoxItem__element, itemClasses]"
+				v-bind="$attrs"
+				@click="$emit('click', $event)">
```

Comment:
isn't this click listener too deep (only on label)? Also structure-wise div > span > **div** by default IMO isn't fine, if we expect inline component, should be span|a|button?

Reply:
Pushed